### DISCUSSION TITLE
Feature to Disable EDP throttling on TUF F15 2022.

### DISCUSSION THREAD

**Discussion by AnujJena** [seerge/g-helper#864] [Ideas]
Hello guys,

ASUS Tuf F15, i7 12700H, RTX 3060 (140W).

While "Plugged in" and gaming the i7 12700H, power consumption drops from 45 to 25 Watts, because of EDP throttling.
This happens only when the charge is not at 100%. And while gaming the charge keeps dropping from 100% even if it's plugged in.

The screenshots below show the average power of CPU and GPU before and after the battery reaches 99%, you can see the battery level in the last column.

The avg clock speed of the CPU decreases as well.

This causes a performance drop in every game significantly.

The CPU Electrical design point becomes "YES" when the battery drops to 99%.
![ws7rwy5otia91](https://github.com/seerge/g-helper/assets/30493525/fe829ccc-b99a-4f8c-83eb-1ae7b8cebbc0)
![fw3apmeptia91](https://github.com/seerge/g-helper/assets/30493525/6a956f2d-4780-4a76-a082-38297d30ecaa)

**Comment by seerge** [comment]
@AnujJena does Armoury Crate allow to control that anyhow? If not - then probably it's hardcoded in bios.

Try ThrottleStop app for finetuning your Intel settings.

**Reply by AnujJena** [reply]
@seerge From my testing, Armoury crate controls the EDP, as uninstalling AC limits the CPU wattage on TUF to 25-35 Watts,
And locks the GPU wattage to 100 Watts.
So, in other words, uninstalling AC disables GPU power of 140 Watts and disables CPU power of 45 Watts.

In the below screenshot, I have changed the AC Initiation settings,
originally it was -
HasBeenUsed=1
ThrottleModeOnAC=1
ThrottleModeOnDC=2

New- 
HasBeenUsed=0
ThrottleModeOnAC=0
ThrottleModeOnDC=1

Changing the ThrottleModeOnAC from 1 to 0, stopped the EDP throttling for a few games.

![Screenshot 2023-07-23 215106](https://github.com/seerge/g-helper/assets/30493525/39faa1b5-c2b9-4ea5-9bdf-45b2e50ac444)

Throttlestop does not work as i7 12700H is a locked CPU.

Moreover when I uninstalled Armoury crate the total power draw of the laptop reduced from 200 Watts to 175 Watts.

I believe, but not sure, that Armoury crate has something to do with the total power draw of the laptop as well.

If you need any screenshots or log files please let me know, I  am happy to help any way I can.

**Reply by seerge** [reply]
@AnujJena well, i don't know if AC sets any "extra" undocumented settings in your specific case. Normally all settings are just part of one of 3 predefined modes (Silent , Balanced, Turbo) as they are baked in BIOS 

Does g-helper show you CPU power sliders under fans+power (it will if it's supported on your device) ? 

If you want you can submit a dump file as explained in the https://github.com/seerge/g-helper/issues/38 , as without physical access to your device i can't add much

**Reply by AnujJena** [reply]
This issue with TUF F15 models with i7 12700H as far as I know, 
A few Reddit users have confirmed that it's the issue with their laptops as well.

Yes, G-Helper does show the CPU power slider, but regardless of the CPU power set,
if the battery charge is anything other than 100% the EDP throttling is enabled, which limits the CPU to 25 Watts.

I have posted the Method 1 output, heres the [link](https://github.com/seerge/g-helper/issues/38#issuecomment-1646901314).

**Reply by seerge** [reply]
@AnujJena I need a second method (with a DSL dump) :) it shows what kind of commands you have available on your laptop.

**Reply by AnujJena** [reply]
@seerge I have created a DSDT.DSL file, how should I upload it?
Apologies for my lack of knowledge here.

**Reply by seerge** [reply]
You can drag-n-drop it to new text message in zip

**Reply by AnujJena** [reply]
@seerge Here it is, 
I hope this is what you needed, XD
 
[dsdt- TUF F15 2022.zip](https://github.com/seerge/g-helper/files/12139812/dsdt-.TUF.F15.2022.zip)

**Reply by seerge** [reply]
@AnujJena hello, thanks for the dump :) I honestly, don't see anything there that would tip me that it's related to all this "throttle" modes you are talking about. It could be some tricky proprietary thing very specific to your model. So without physical device and lots of tinkering it will be impossible to figure it out. 

Does lowering charge rate  to 80% remove this throttling thing if you go below 100% ?

**Reply by AnujJena** [reply]
@seerge Hello, and thank you for looking into this for me, I really appreciate it,
Lowering the charge below 80% or lower has no change.
However, after changing the ThrottleModeOnAC to 0, the CPU Throttling gets disabled for some games, while it stays on for some games. 
Its really weird.


However, what stays constant is a 200 Watt power limit for CPU+GPU+Misc.
Was there anything in the dump indicating a 200 Watts power cap?

However, what stays constant is a 200 Watt power limit for CPU+GPU+Misc.
Was there anything in the dump indicating a 200 Watts power cap?

As when the CPU is at 45 Watts, the GPU maxes at 115 Watts, and when the CPU stays at 35 Watts the GPU boosts upto 125 Watts.

**Reply by IceStormNG** [reply]
It could be that this setting configures Intel Dynamic Tuning (Intel Innovation Platform). This is present on any Intel laptop and will dynamically change PL1/PL2 and dynamic boost of the GPU depending on some settings in a ACPI Table that is hidden somewhere and can only be changed through some "magic" API that is not publicly documented.

**Reply by AnujJena** [reply]
Yeah, it seems to be the Dynamic Tuning changing the power.
But the I still havent found what causes the TUF F15 model to have a 200 Watt power limit, 
because when I tested the same games on a Lenovo Legion 5i and ROG M16, with the same specs, 
those laptops had 220 Watts power where as TUF F15 gets locked at 200 Watts, 
its not even thermal throttling.

**Comment by kuonji-arisu** [comment]
This is a well-known issue for Chinese TUF users.
When the battery is not fully charged, ASUS has limited the total power consumption of CPU+GPU to 160 watts.

Unfortunately, there is currently no known solution to fix this problem. Many users have reported the issue to ASUS, who promised to release a BIOS update to address it. However, it appears they have yet to follow through on that promise and may never do so.

Thankfully, there are a few methods that can help alleviate the issue.

1. One option is to use "Standard mode" and connect an external monitor via DP/HDMI, as the overall power restriction only applies in dGPU mode.
2. Another option involves turning off all e-cores, unlocking bios advanced options, and disabling Intel Dynamic Tuning and Nvidia Dynamic Boost. After doing so, limiting CPU power (PL1/PL2) to 45w will allow for stable CPU/GPU power at 45+115w for an extended period.

**Reply by XyKER1** [reply]
How do you disable Dynamic Tuning and Nvidia Dynamic Boost?  Also, I know there is an advanced BIOS menu, but how do you "unlock" the advanced options?  I'm having the same issue with my TUF F15 2022 (except CPU+GPU limited to 140 watts instead of 160 watts for you), but I'm not sure how to change these.  Any help would be greatly appreciated, as this has bothered me for months!

**Reply by IceStormNG** [reply]
Dynamic tuning can be disabled by uninstalling the dynamic tuning device drivers and adding a registry tweak to prevent re-install.

Advanced bios cannot be unlocked anymore due to intel boot guard. Any bios modification will more or less brick the firmware.
As for dynamic boost... I'm not sure whether this is just driver or whether it is embedded into the EC.

**Reply by XyKER1** [reply]
So you're saying this person's 2nd fix won't work?  That's disappointing.  However, I have made some modifications to the BIOS (e.g. unlocking turbo ratio limits and modifying IMON slope) and it hasn't bricked the BIOS.  Maybe there are some other settings (e.g. I've seen one called ICC Max Unlimited Mode) that will fix the throttling.  I guess I'll start testing out some of those settings sometime soon and report any findings here.

**Reply by IceStormNG** [reply]
Did you modified the BIOS image and re-flashed it? Or have you written to the configuration to unlock the setting?

Bootguard is also not enabled on all laptops. On most newer laptops (11th gen and newer) it is enabled.

**Reply by XyKER1** [reply]
Yes, I wrote to the configuration using grubx64efi.  This program allows you to change the values of BIOS variables without disabling bootguard (although I did disable secure boot).  If you are curious about how I did this, I can give you more information.

**Reply by XyKER1** [reply]
Also, how do you uninstall Dynamic Tuning?  All I see in device manager is "Intel Dynamic Tuning Technology Device Extension Component," nothing else relates to it.  Disabling that doesn't affect the throttling.

**Reply by IceStormNG** [reply]
There are multiple devices in the "System device" section. Like Intel Dynamic Tuning Participant, and also other devices. Sometimes it's also called "Intel Innovation Platform"

**Reply by AnujJena** [reply]
Can you please share how to do it.

**Reply by XyKER1** [reply]
I see Intel Innovation Platform, so I'll try disabling that and see if it works.  Nothing about Dynamic Tuning, though.

@AnujJena I will write out a list of steps, might be a couple hours till I get around to it since it's fairly complicated.

**Reply by AnujJena** [reply]
@XyKER1 Thank you, I appreciate it.
And btw, I am seeing multiple Intel Innovation Platforms in Device Manager, which one should I disable?

![image](https://github.com/seerge/g-helper/assets/30493525/d387e4d0-7dc3-4c66-83d7-8f6bf5b43ba1)

**Reply by IceStormNG** [reply]
For me, the dynamic TDP adjustment stopped when I disabled that one 
<img width="353" alt="image" src="https://github.com/seerge/g-helper/assets/12786283/2e18cc7b-e2b4-45c9-8a15-99ae1f3ad7ff">


Though, this locked my CPU to 30W, but throttlestop could then freely increase it. EDP Throttling still happens though. Afaik, EDP throttling is hardware imposed either by the CPU or by the EC. The bios can configure a current limit for the VRMs which might be set too low by ASUS (probably intentional).

**Reply by AnujJena** [reply]
So after increasing CPU Wattage using throttlestop, does it increase the CPU wattage, even if EDP is enabled?

**Reply by IceStormNG** [reply]
It allows for more wattages, yes, but the EDP Throttle still pops up all the time in "Limits". Though for me, it can keep the clock up until it hits 95°C throttle temp, but I also have a ROG M16, which is a different machine.

**Reply by AnujJena** [reply]
Ah, for ROG M16, I think you have USB C charging as well right?
And it can be used along side the barrel charger for extra power.

**Reply by IceStormNG** [reply]
Afaik, no. It only uses one power source (at least from my testing), but the 280W brick has enough power to not get power limited by that. The machine is usually just thermally limited as it is rather thin (for what they put into it).

**Reply by XyKER1** [reply]
@AnujJena Here's how to modify BIOS variables.  Please know that I haven't tested out any variables that could fix this issue yet, so I'm not guaranteeing this will fix the issue.  I think there is a good chance this could be used to fix the EDP throttling, though.  Also, I got most of the information from this video: https://www.youtube.com/watch?v=0gMmfexcjNs

First of all, download [Universal Bios Extractor 2.0](https://github.com/zabiyakod/Bios-Reader/raw/master/Universal%20BIOS%20Backup%20ToolKit%202.0.exe), [IFR Extractor](https://github.com/LongSoft/Universal-IFR-Extractor/releases/download/v0.3.6/IRFExtractor_0.3.6_win.zip), [UEFI Tool](https://github.com/LongSoft/UEFITool/releases/download/A67/UEFITool_NE_A67_win64.zip), and grubx64.efi (see the Google Drive link in the YouTube video's description for this).

Then, get a 16 GB or smaller flashdrive and format it to FAT32.  I think you can also do this by creating a FAT32 partition on a larger storage device.

Open Universal Bios Extractor 2.0, check the box next to custom, and change the dropdown to the amount of RAM you have divided by 1000 (probably 16384K if you have the TUF F15 2022).  If you have more RAM than that, just select 16384K.  Select read, wait for it to finish, then select backup to save the file to somewhere.

Open UEFITool and click file, open, select the BIOS file you just extracted.  Then do Ctrl + F to search, change tab to text, and search for "iccmax."  It should show some matches in Setup/PE32 (in the white box at the bottom).  Double click on one of these results and it will take you to this location in the file tree above.  A bar called "PE32 image section" should be selected.  This will be a subheading under a section called "Setup."  Left click on "Setup" to select it, right click on this section, extract as is, then save the .ffs file to anywhere.

Open IFR extractor, click on the three dots to select the .ffs file.  Select extract and save the resulting .txt file anywhere.

Next is the tricky part.  This .txt file contains all the advanced BIOS options relating to CPU control that are hidden.  A lot of the file will be gibberish, but you'll see some names of settings you can change.  For example, search up "overclocking lock."  The result is an example of a BIOS variable.  You need to look at the value after the text "VarStoreInfo (VarOffset/VarName)."  For overclocking lock, this is 0x10E for me.  Basically you'll have to scroll through this entire document and pick out variable names that seem like they might fix the throttling.  Here's a list of ones I've found that seem significant (I haven't modified any of their values yet):
ps current threshold
CS PL1 Limit
ia icc unlimited mode
vccin aux icc max
configtdp [nominal, level1, level2] power limit 1
ring voltage mode - override
vr current limit
imon offsets
pch cross throttling
ec turbo control mode
turbo mode
ctdp bios control
p-state capping
platform pl1
power limit 1 override
package power limit msr lock
power limit 4
power limit 3
speedstep and speedshift
hwp options
battery participant
powermeter
[enhanced] thermal velocity boost
overclocking feature

The last part is to actually modify these variables.  You'll need to write down the VarStoreInfo of anything you're modifying somewhere else.  **For this case, I'll just talk about how to do this for "overclocking lock."**  (fyi this doesn't actually allow overclocking or undervolting, but if you disable this along with CFG lock, you can change the turbo ratio limits).  I would recommend trying to modify this setting for your first time to make sure you know what you're doing (it can't break anything no matter what value you set it to).  Note the Min and Max values further down the line in the .txt document from VarStoreInfo.  These tell you the minimum and maximum values these variables can be set to.  In this case, min is 0x0 and max is 0x1 (the "0x" denotes a hexadecimal number, but the actual value you are setting it to is what's after the "0x").  0 is disabled while 1 is enabled.

I want to disable overclocking lock by setting the value of it to 0x0.  REMEMBER: all values are in hexadecimal.  YOU WILL NEED TO CONVERT YOUR DESIRED VALUE FOR THE VARIABLE FROM DECIMAL TO HEXADECIMAL before writing it to the BIOS.  There are plenty of online converters than can do this.  Remember to write down the VarStoreInfo value on a piece of paper or another device before continuing.  For me (and probably you), this would be 0x10E (another example: for CFG lock it would be 0x43).  Also write down the value you want to set this variable to. 0x0 for this case.

Plug in your FAT32 flashdrive, make sure it is empty, and make a folder called "EFI" (everything inside the quotations).  Create a folder inside of "EFI" and call it "Boot".  Then, download grubx64.efi and move it into the "Boot" folder.

Next, restart your PC.  You can then go to the boot menu (by pressing Esc when rebooting) and move the flashdrive to the top.  This will be a one time change, so next time you turn on the laptop, it will revert to the Windows drive.  Alternatively, you could go into the Bios and move the flashdrive to the top of the boot list.

You will then boot into the flashdrive.  To modify BIOS variables, use the following syntax (without brackets):
setup_var CpuSetup [VarStoreInfo] [desired value]
For example, to disable overclocking lock, you would type this and then press enter:
setup_var CpuSetup 0x10E 0x0

Note that after pressing enter, it will tell you the current value of the variable (in this format: "Offset [VarStoreInfo] is [current value]"), then tell you it's writing the desired value to the variable.  If it says "Writing [desired value] to [VarStoreInfo]", this means it successfully changed the value to what you wanted.  However, if you used a number that is too high or low, I'm guessing it would be reset to default next time you boot the laptop.  After modifying all the variables you want to, press the power button to turn the laptop off.  Turn it back on and if you change the boot order in the BIOS, you will have to put the Windows drive back on top.  You will then boot back into Windows.

If you have any questions about this, let me know.  Remember, the YouTube video linked at the top essentially shows you how to do this for "overclocking lock" and "CFG lock", so watch that first if you are confused about a step.  I wrote this guide to explain more in depth what he's doing so you could understand how this would be done for other settings that might fix the CPU throttling.

**Reply by kuonji-arisu** [reply]
**only for tuf 15 intel model (FX507ZM)**
1. disable security boot control
2. copy grubx64.efi to the EFI part
3. create an EFI boot option ( path to grubx64.efi on Disk0 )
4. boot to grub
5. in grub CLI, input `setup_var AsusHidePageVariable 0 01`
6. reboot to bios, then you can see some extra settings start with crb_  in advanced settings
7. crb advanced->thermal configuration->IDT Configuration->Disabled
[grubx64.zip](https://github.com/seerge/g-helper/files/12210678/grubx64.zip)

**Reply by kuonji-arisu** [reply]
for dynamic boost, you can turn off that by disabling this in the device manager
![image](https://github.com/seerge/g-helper/assets/29818319/4f8ff584-b18b-42c4-a56c-1b00c99c9422)

**Reply by kuonji-arisu** [reply]
You don't necessarily need to shut down IDTF; it is optional. Simply disabling e-cores and limiting CPU consumption can bring some improvement to online games.

Of course, disabling e-core would hurt productivity applications like video editing or programming.
Additionally, EDP throttling on TUF15 would only affect multiplayer games. If you only play large 3A games, there is no need to perform these actions.

Finally, there is an option in crb advanced->platform settings->charging method->fast charging. I'm not sure if this option works, but you can give it a try.

**Reply by XyKER1** [reply]
Thanks, I will try this tomorrow and see if it stops the EDP throttling!

**Reply by kuonji-arisu** [reply]
It cannot "stop" EDP throttling; This is hard coded into the EC, and only ASUS has the capability to modify it.
It merely allocates the power consumption of the CPU/GPU to a reasonable value to ensure the long-term stability of FPS during multiplayer gaming.
When the battery is not fully charged (less than 100%), the overall power consumption of the system is still limited to 160 watts. The only difference is that it changes from 20+90w (hitting the power consumption wall) to 45w+115w.

**Reply by AnujJena** [reply]
Disabling Nvidia platform controllers and frameworks limits the GPU power to 100 Watts.

**Reply by kuonji-arisu** [reply]
Yes, it's just a redistribution of power. Disabling DB is more suitable for long-time multiplayer online gaming, where these games typically require more CPU power. For multiplayer games, limiting the CPU power to 20 watts is unacceptable.

**Reply by XyKER1** [reply]
Yeah so I tried your method and it gives the CPU 15 more watts (up to 40 watts, not 45), which is still a substantial improvement over 25.  It still looks like there is no way to increase CPU+GPU power past 140 watts, as ASUS knows this will cause battery drain.  I tried changing ThrottleModeonDC to 0, which I think would actually fix this issue (as it probably stops throttling when battery is being used - which happens on this laptop while total draw is above 140 watts), and I got frequent graphical bugs and crashes (probably meaning the GPU isn't getting enough power).  ThrottleModeonAC doesn't change anything for me.  I'll going to keep trying to fix this but I have a feeling there won't be a way to get long term power draw up to the 160 W it can sustain while at 100% battery.

**Reply by 1m4g1ne** [reply]
I think it's not simply just battery drain , look at the pics I sent below , even after water cooling , CPU can still gets to 66C , under normal condition , both CPU and GPU are under 60C if water cool. If this is done without water cool , I am pretty sure , CPU would reach throttle temp of 100C.

**Reply by AnujJena** [reply]
Disabling Dynamic boost enables the CPU to draw 45 Watts, but limits the GPU below at 100 Watts.
Is there any possible way to re-enable GPU boost?

And it is really weird that disabling Intel Dynamic boost, limits the GPU Boost.

**Reply by IceStormNG** [reply]
It is not weird at all. Nvidia Dynamic Boost and Dynamic Tuning work hand in hand.
The Dynamic Boost works by relocating unused CPU Power budget to the GPU. For this to work, Dynamic Tuning tries to predict how much CPU load and how much GPU load is running to prioritize either GPU or CPU. If Dynamic Boost does not free up power budget from the CPU, the GPU cannot take it.
Without dynamic tuning, the GPU is locked to its TGP and cannot boost.

**Reply by AnujJena** [reply]
@IceStormNG I see, it makes sense.

**Comment by 1m4g1ne** [comment]
About this , I recently modified my Tuf F15 2022 with water cooling , if i done testing water cooling I wanted to clean install windows and make a fresh start with laptop. But before the clean install , I was using the laptop for about a week. I have some drivers from TUF F15 2023 installed ( since one varient of TUF F15 2023 [FX507ZV] comes with i7-12700H). I played games like Valorant and i saw CPU is using 60W and GPU is using 135W , total power comsumption is so high , even at plug-in state , my laptop battery down from 100 to 68. I thought CPU is capable of using so much power because of low temperature from water cooling. 

Then I did a clean-install of windows, and i still install some of TUF 15 2023 (FX507ZV) drivers , but the total power consumption is back to normal , even with water cooling enabled , CPU is only using about 30W and GPU about 110-125W. I am still trying to replicate that high total power consumption like last time , but still not successful.

**Reply by AnujJena** [reply]
Which drivers had you installed from the 2023 Model?

**Reply by 1m4g1ne** [reply]
![Screenshot (26)](https://github.com/seerge/g-helper/assets/112641970/b5cbf655-0412-457f-8432-f17d1d7f76eb)
All driver above , but installing these driver doesn't unlock the wattage , after clean install windows i still have these drivers installed , but still don't trigger wattage unlock like last time. I am still finding a way to replicate that event.

**Reply by XyKER1** [reply]
Have you installed Armoury Crate and Nvidia Geforce Experience yet?

**Reply by 1m4g1ne** [reply]
No, I totally debloat my laptop , only using ghelper and use nvcleanstall to only install minimum required parts.

**Reply by 1m4g1ne** [reply]
YES!!!!!!!!!!!!!!! hello i finally replicate the event!!!!

**Reply by 1m4g1ne** [reply]
![Screenshot (27)](https://github.com/seerge/g-helper/assets/112641970/bfa2989e-9b94-4223-92e8-2ce336006df0)
![Screenshot (28)](https://github.com/seerge/g-helper/assets/112641970/ff479340-bd84-4f5a-9f99-0d9d815b95a7)
![Screenshot (29)](https://github.com/seerge/g-helper/assets/112641970/33bc2fc7-d2d4-4bd7-b2f6-1d2c7d933126)
![Screenshot (30)](https://github.com/seerge/g-helper/assets/112641970/85879272-e90d-4d1c-a8d9-fee0cb8b8d1e)

Able to reach CPU 60W + GPU 110W.
High exposure from pics are from HDR.

**Reply by XyKER1** [reply]
How did you replicate it?  I've found this will happen for me while the battery is at 100% or occasionally for about 10 seconds without doing any of the changes mentioned in this thread, but are you able to sustain it once your battery drops to 99%?

**Reply by 1m4g1ne** [reply]
My battery isn't even fully charged.
![Screenshot (31)](https://github.com/seerge/g-helper/assets/112641970/cf720439-8308-468e-9f7f-d2e062c3550e)
With FX507ZV drivers installed.
In ghelper i set CPU boost to agressive , in windows setting power & battery , I set it to best performance.
And another reason can be because of water cooling , my CPU temps are always below 70C.
I also did this before i modify my laptop for watercooling , i don't remember seeing such power draw on air cool.
But yea you can try it out.

**Reply by XyKER1** [reply]
So you downloaded and installed every driver from this page, correct? https://www.asus.com/us/supportonly/fx507zv/helpdesk_download/

**Reply by 1m4g1ne** [reply]
Yea from that page but not every , just like photos above , only install intel CPU related drivers.

**Reply by 1m4g1ne** [reply]
![Screenshot (32)](https://github.com/seerge/g-helper/assets/112641970/b4209356-3f4f-4fa0-b85c-aed20bf3aed3)
Here , I tested with water cooling off , CPU+GPU total Pwr is locked at 130-140W.
With water cooling it can go as high as 175W.
So just by cooling temps down you can get CPU to use more power without unlocking anything.

**Reply by XyKER1** [reply]
That's very strange.  Maybe there is some thermal trip point in combination with less than 100% battery that causes CPU to throttle to 25 W.  I saw something about a "thermal trip point" at 71 C in the "thermal configuration" section of the advanced BIOS that @kuonji-arisu showed me how to unlock.  Maybe since your temps with water cooling are below that it doesn't throttle the power?  I'm not throttling even at 45+115 W (I use a cooling pad), but my temps are above 71 C.

**Reply by AnujJena** [reply]
For me everythings going to shit.
Dynamic Tuning disabled, Nvidia framework disabled, AC throttle config- 0,
Still the CPU throttles down to 32 watts now. 

It was all running okay till a windows update f-ed it all up.

**Reply by XyKER1** [reply]
Nvidia framework re-enables everytime you restart your PC, so are you sure it's still disabled?  Also, check your CPU frequency.  Some games rely on single-core performance, and if that CPU core is at its max speed (probably 4.1 ghz due to turbo ratio limits), the CPU power draw won't go any higher.  Also, your GPU might be at 100% usage.

**Reply by AnujJena** [reply]
@XyKER1 I reset Windows and reinstalled drivers.
And now after disabling DT and NFW, the CPU Wattage stays at 45 Watts and GPU at 100 Watts. 

Btw, how did you manage to achieve 45 CPU and 115 GPU? 
Was the BIOS modding?

**Reply by XyKER1** [reply]
No, that was just at 100% battery (within 10 minutes at 45+115, it drains to 99% and power drops to 40+100).  If you read @kuonji-arisu's comment on the other thread, he talks about how to show all the hidden settings I was talking about right in the BIOS, making them much easier to change than my method.  I might experiment with them if I get bored sometime.

**Reply by 1m4g1ne** [reply]
![IMG_20230803_195739](https://github.com/seerge/g-helper/assets/112641970/21fc56f2-85ab-46be-9a0a-4c649c155b40)
Yea that's strange , without water cooling , once battery drop from 100% it lower the wattage , but when temperature is cool enough power draw is still high even when battery is in charging mode. But it won't charge , wattage is so high is just draw power from battery instead.

The pic is TUF F15 2022 btw , I just modifed mine to 16inches and watercooling.

**Reply by AnujJena** [reply]
ASUS's decision not to have pass-through charging is criminal.

**Reply by IceStormNG** [reply]
Every laptop made in the last century has pass-through charging. Some laptops can consume more power under peak load than the PSU can provide. So they pull the remaining power from the battery. And I guess ASUS uses that as a sign to power limit the machine to prevent the machine from emptying the battery. That's at least my assumption.

From what I've seen online, the TUF F15 comes with a 200W PSU. That might not be enough for peak load spikes. Keep in mind that CPU package power is ... "guessed" and not measured. It is computed by VID and some parameters. The CPU has not wattage sensor that accurately reads the power it draws. I guess the same is true for the GPU.
Also keep in mind that there are other components that draw power and VRMs have loss.

**Reply by AnujJena** [reply]
The 240 Watts charger which comes with the laptop or even the 280 Watts ROG charger both are capable enough.
And passthrough charging should by-pass the battery, but I am saying that TUF does not have pass-through charging because I have tried to run the laptop after removing the battery, i.e. I unplugged the battery and plugged in the 280 Watt charger.

What happened is, the CPU was locked to 15 Watts and the GPU got disabled. LOL.

I am guessing that if it would have been pass-through, the laptop would have functioned without any power limitations with the battery removed.

**Reply by IceStormNG** [reply]
This is not because it has no passthrough charging. Some OEMs trigger BD Prochot when the battery is not installed as they know that the PSU might not provide enough power for the system under peak load.
MacBooks do the same. They lock the CPU to 800Mhz if not battery is installed. Has nothing to do with pass through charging.

**Reply by AnujJena** [reply]
Okay, makes sense.

**Reply by AnujJena** [reply]
@IceStormNG Going off topic for a bit, can you help me deciding which RTX 4060 laptop would be a good choice?
A laptop that will not have an EDP throttling like the TUF Intel series.

Would TUF A15 2023 be a good model?

**Reply by IceStormNG** [reply]
All models have EDP Throttle. It is the CPU that does that mainly + BIOS current limitations set by EC or VRs. I have no experience with the TUFs though, I always went with the Zephyrus because I like their design and the portability + performance. I only game a little bit here and there.

**Reply by XyKER1** [reply]
@AnujJena I watch Jarrod's Tech (a gaming laptop reviewer) a lot, and without a doubt, the best models other than the TUF series are Lenovo Legion or Acer Predator Helios.  They're well priced, upgradeable, and don't have any major drawbacks.  You could go with the TUF A15, but the other two, while maybe slightly more expensive, have better cooling and you don't run the risk of extreme EDP throttling.

**Reply by XyKER1** [reply]
@1m4g1ne Does your battery discharge while playing games with your water cooling (and fully unlocked CPU), or does it stay constant?  I finally got a new game that's heavily CPU limited, so I've decided to start attempting some BIOS modifications to fix the throttling issue.  This information might save me a lot of time.

**Comment by XyKER1** [comment]
Just wanted to post a final update on this situation. Two weeks ago, my TUF F15's charger broke, as it kept disconnecting and reconnecting, making me unable to do anything more intensive than light web browsing. I sent it in for repair under an extended warranty and they gave me a full refund as they were unable to fix it. I bought a Lenovo Legion with very similar specs (i7-13700H and 4060), which is so much better in every way than the TUF F15 (with the exception of the touchpad being garbage). However, it came with a 230 W adapter, and whenever the CPU+GPU power draw is above 135 W, the battery starts draining while gaming, although I can still get 70+ W on the CPU (compared to 25 W on the TUF). However, many people on Reddit have figured out that buying the 300 W adapter will stop the battery from draining.

With this in mind, I can conclude that the reason the TUF was throttling was because the adapter is not powerful enough to supply the components at 45+115 W. You can't just add up the power draws and compare them to the total adapter wattage - power supplies don't work like that.

I also had been doing an enormous amount of BIOS tinkering over the last few months, and nothing allowed the battery to drain further to improve performance.

So, the only possible way to stop this throttling is to buy the 330 W charger for ROG laptops, which looks like it has the same port shape and should work.

**Comment by Kikimahe** [comment]
Ooooh... I just discovered this topic and I have the same problem on my F17 (2022) 12700H-RTX 3060. [#1574](https://github.com/seerge/g-helper/discussions/1574)

With Ghelper I had to limit my CPU to 40W and the GPU to 1700Mhz (95W-100W) to obtain acceptable performance and not exceed the throttling limit.

Thanks for sharing all of this.

**Reply by XyKER1** [reply]
I'm Raibex87 on the ROG forum and I can tell you that you don't need to read everything here (I already have). Nothing will fix this issue other than getting a new power adapter (and you can limit the GPU clock using G-Helper to allow higher CPU power draw, like you said).

**Reply by Kikimahe** [reply]
Oh Raibex87 ! :-) 

Yes, I see that nothing can solve the problem. :(

I even tried reinstalling Armory Crate this afternoon (shame on me). 
But obviously that didn't change anything and I immediately uninstalled it.

For buying a new power adapter, I looked at the price and it's around €200 ... Ouch.
And we are not 100% sure that fix it. ;'(

**Reply by XyKER1** [reply]
@Kikimahe The only website I'm aware of that sells official ASUS spare parts is A-Accessories, although it looks like that adapter is sold out for them. Not sure why it's so expensive for you - for me pretty much every charger on Amazon is $100 or less, although I do live in the US. I would recommend against buying from Amazon if that's what you were looking at, as aftermarket chargers are very unreliable.

I do see some listings for $45 US dollars on Ebay which say brand new OEM 330 W ROG charger with 30-day returns, so maybe there are some deals like that available in Europe.

I know it's a pain, but I'm 99.9% sure that nothing else will work considering a laptop from a completely different brand has the exact same battery drain issue with a similar CPU+GPU wattage to total adapter power ratio (135:230 on my Legion, 146:240 on the TUF). The 240 W adapter is simply not powerful enough, and even if you were able to find a way to allow battery drain, that would destroy your battery capacity rather quickly if it's draining every time you're playing games.

**Reply by XyKER1** [reply]
@Kikimahe Actually ignore what I said before, I just got the 300 W power adapter, and it doesn't make a difference. Even though my Legion Slim 5i Gen 8's thermal system can handle a lot more power, it looks like the motherboard caps the input to 230 W, so I wouldn't be surprised if ASUS did that, too. I'm going to return this and get a Legion Pro 5i which actually supports the 300 W adapter, so hopefully that won't have the battery drain issue. With that being said, there may actually not be a way to stop the EDP throttling on the TUF.

**Reply by Kikimahe** [reply]
I was thinking of buying a 330W adapter soon...
Thanks for your feedback, I wouldn't do it.  

For the moment I keep the battery charged at 100% to maintain maximum performance all the time. I will replace the battery when it dies.
And when I play games, I pray that it doesn't go down to 99% :-D
